xmm/analysis-xmm-epic-reprocessing.md

```diff
@@ -0,0 +1,403 @@
+- filtered_event_list: File name of the output file, or filtered event list.
+
+```python
+def apply_simple_filter(unfiltered_event_list,mos=True,pattern=12,
+                          pi_min=200,pi_max=12000,flag=None,
+                          filtered_event_list='filtered_event_list.fits'):
+    
+    if flag is None:
+        if mos:
+            flag = '#XMMEA_EM'
+        else:
+            flag = '#XMMEA_EP'
+    else:
+        flag = '(FLAG == {0})'.format(flag)
+    
+    # "Standard" Filter
+    expression = "expression='(PATTERN <= {pattern})&&(PI in [{pi_min}:{pi_max}])&&{flag}'".format(pattern=pattern,pi_min=pi_min,pi_max=pi_max,flag=flag)
+    
+    inargs = ['table={0}'.format(unfiltered_event_list), 
+              'withfilteredset=yes', 
+              expression, 
+              'filteredset={0}'.format(filtered_event_list), 
+              'filtertype=expression', 
+              'keepfilteroutput=yes', 
+              'updateexposure=yes', 
+              'filterexposure=yes']
+    
+    w('evselect', inargs).run()
+```
+
+The second function will create a FITS image file and plot it. The inputs are:
+
+- event_list_file: File name of the event list to be plotted.
+- image_file: Name of the output FITS image file.
+
+```python
+def make_fits_image(event_list_file, image_file='image.fits'):
+    
+    inargs = ['table={0}'.format(event_list_file), 
+              'withimageset=yes',
+              'imageset={0}'.format(image_file), 
+              'xcolumn=X', 
+              'ycolumn=Y', 
+              'imagebinning=imageSize', 
+              'ximagesize=600', 
+              'yimagesize=600']
+
+    w('evselect', inargs).run()
+
+    hdu = fits.open(image_file)[0]
+    wcs = WCS(hdu.header)
+
+    ax = plt.subplot(projection=wcs)
+    plt.imshow(hdu.data, origin='lower', norm='log', vmin=1.0, vmax=1e2)
+    ax.set_facecolor("black")
+    plt.grid(color='blue', ls='solid')
+    plt.xlabel('RA')
+    plt.ylabel('Dec')
+    plt.colorbar()
+    plt.show()
+```
+
+In the cell below we will range over all event lists from the three EPIC instruments (EPIC-pn, EPIC-MOS1, EPIC-MOS2). An image file will be created from each event list and a plot will be made.
+
+```python
+# For display purposes only, define a minimum filtering criteria for EPIC-pn
+
+pn_pattern   = 4        # pattern selection
+pn_pi_min    = 300.     # Low energy range eV
+pn_pi_max    = 12000.   # High energy range eV
+pn_flag      = 0        # FLAG
+
+# For display purposes only, define a minimum filtering criteria for EPIC-MOS
+
+mos_pattern   = 12      # pattern selection
+mos_pi_min    = 300.    # Low energy range eV
+mos_pi_max    = 20000.  # High energy range eV
+mos_flag      = None    # FLAG
+
+os.chdir(odf.work_dir)
+
+pnevt_list = odf.files['PNevt_list']
+m1evt_list = odf.files['M1evt_list']
+m2evt_list = odf.files['M2evt_list']
```

Comment:
In a future release of pySAS it will be changed to `PNevt_list`, `M1evt_list`, `M2evt_list` to make things internally consistent with other things in SAS. It's not a big deal now, but I have already made the change in the pySAS repository.

Fixed for now.